plexapi/base.py

```diff
@@ -812,6 +820,75 @@ def updateTimeline(self, time, state='stopped', duration=None):
         self._reload(_autoReload=True)
 
 
+class PlexSession(object):
+    """ This is a general place to store functions specific to media that is a Plex Session.
+
+        Attributes:
+            live (bool): True if this is a live tv session.
+            player (:class:`~plexapi.client.PlexClient`): PlexClient object for the session.
+            session (:class:`~plexapi.media.Session`): Session object for the session
+                if the session is using bandwidth (None otherwise).
+            sessionKey (int): The session key for the session.
+            transcodeSession (:class:`~plexapi.media.TranscodeSession`): TranscodeSession object
+                if item is being transcoded (None otherwise).
+    """
+
+    def _loadData(self, data):
+        self.live = utils.cast(bool, data.attrib.get('live', '0'))
+        self.player = self.findItem(data, etag='Player')
+        self.session = self.findItem(data, etag='Session')
+        self.sessionKey = utils.cast(int, data.attrib.get('sessionKey'))
+        self.transcodeSession = self.findItem(data, etag='TranscodeSession')
+
+        user = data.find('User')
+        self._username = user.attrib.get('title')
+        self._userId = utils.cast(int, user.attrib.get('id'))
+        self._user = None  # Cache for user object
+
+        # For backwards compatibility
+        self.players = [self.player] if self.player else []
+        self.sessions = [self.session] if self.session else []
+        self.transcodeSessions = [self.transcodeSession] if self.transcodeSession else []
+        self.username = [self._username] if self._username else []
```

Comment:
To retain backwards compatibility:
```suggestion
        self.usernames = [self._username] if self._username else []
```